### PR TITLE
feat(frontend): Add main marketplace page tests and page object structure

### PR DESCRIPTION
**Pull request by Abhi1992002**
- Resolves - https://github.com/Significant-Gravitas/AutoGPT/issues/10426
- Need to review this pr, once this issue is fixed - https://github.com/Significant-Gravitas/AutoGPT/issues/10404

I’ve created additional tests for the main page, divided into two parts: one for basic functionality and the other for edge cases.

**Basic functionality:**
- Users can access the marketplace page when logged out.
- Users can access the marketplace page when logged in.
- Featured agents, top agents, and featured creators are visible.
- Users can navigate and interact with marketplace elements.
- The complete search flow works correctly.

**Edge cases:**
- Searching for a non-existent item shows no results.

### Changes

- Introduced a new test suite for the marketplace, covering basic functionality and edge cases.
- Implemented the MarketplacePage class to encapsulate interactions with the marketplace page.
- Added utility functions for assertions, including visibility checks and URL matching.
- Enhanced the LoginPage class with a goto method for navigation.
- Established a comprehensive search flow test to validate search functionality.

#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan
- [x] I have tested my changes according to the test plan:
  - [x] I have done all the tests and they are working perfectly

### CODE REVIEW

File: autogpt_platform/frontend/src/tests/marketplace.spec.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    await firstFeaturedAgent.click();
+    await page.waitForURL("**/marketplace/agent/**");
+    await matchesUrl(page, /\/marketplace\/agent\/.+/);
+    await marketplacePage.goto();
+
+    const firstTopAgent = await marketplacePage.getFirstTopAgent();
+    await firstTopAgent.click();
+    await page.waitForURL("**/marketplace/agent/**");
+    await matchesUrl(page, /\/marketplace\/agent\/.+/);
+    await marketplacePage.goto();
```
Comment (0ubbe) [inline autogpt_platform/frontend/src/tests/marketplace.spec.ts]: These two assertions are very similar. Some UI or text on the featured agent and top agent pages that could helps us distinguish and assert is the correct view?

**Review by 0ubbe — APPROVED** [review]
Very nice 🙌🏽

**Review by majdyz — APPROVED** [review]

**Review by alaineid — APPROVED** [review]

File: autogpt_platform/frontend/src/tests/marketplace.spec.ts (hunk 1 above)
Comment (Abhi1992002) [reply autogpt_platform/frontend/src/tests/marketplace.spec.ts]: Both top and featured agents will navigate to the same page. The difference between these agents is that they are placed in different sections on the marketplace, that's it.

**Review by 0ubbe — APPROVED** [review]
